packages/back-end/src/services/experimentNotifications.ts

```diff
@@ -0,0 +1,232 @@
+  context,
+  experimentId,
+}: {
+  context: Context;
+  experimentId: string;
+}) => {
+  const experiment = await getExperimentById(context, experimentId);
+
+  if (!experiment) throw new Error("Error while fetching experiment!");
+
+  if (!experiment.pastNotifications?.includes("auto-update")) return;
+
+  await dispatchEvent(context, {
+    type: "auto-update",
+    success: true,
+    experimentId,
+    experimentName: experiment.name,
+  });
+
+  await updateExperiment({
+    experiment,
+    context,
+    changes: {
+      pastNotifications: (experiment.pastNotifications || []).filter(
+        (v) => v !== "auto-update"
+      ),
+    },
+  });
+};
+
+export const MINIMUM_MULTIPLE_EXPOSURES_PERCENT = 0.01;
+
+const notifyMultipleExposures = async ({
+  context,
+  experiment,
+  lastResult,
+  snapshot,
+}: {
+  context: Context;
+  experiment: ExperimentInterface;
+  lastResult: ExperimentReportResultDimension;
+  snapshot: ExperimentSnapshotDocument;
+}) =>
+  updateWrapper({
+    context,
+    experiment,
+    type: "multiple-exposures",
+    handler: async () => {
+      const totalsUsers = lastResult.variations.reduce(
+        (totalUsersCount, { users }) => totalUsersCount + users,
+        0
+      );
+      const percent = snapshot.multipleExposures / totalsUsers;
+      const multipleExposureMinPercent =
+        context.org.settings?.multipleExposureMinPercent ??
+        MINIMUM_MULTIPLE_EXPOSURES_PERCENT;
+
+      if (snapshot.multipleExposures < multipleExposureMinPercent) return;
+
+      await dispatchEvent(context, {
+        type: "multiple-exposures",
+        experimentId: experiment.id,
+        experimentName: experiment.name,
+        usersCount: snapshot.multipleExposures,
+        percent,
+      });
+    },
+  });
+
+export const DEFAULT_SRM_THRESHOLD = 0.001;
+
+const notifySrm = async ({
+  context,
+  experiment,
+  lastResult,
+}: {
+  context: Context;
+  experiment: ExperimentInterface;
+  lastResult: ExperimentReportResultDimension;
+}) =>
+  updateWrapper({
+    context,
+    experiment,
+    type: "srm",
+    handler: async () => {
+      const srmThreshold =
+        context.org.settings?.srmThreshold ?? DEFAULT_SRM_THRESHOLD;
+
+      if (srmThreshold <= lastResult.srm) return;
+
+      await dispatchEvent(context, {
+        type: "srm",
+        experimentId: experiment.id,
+        experimentName: experiment.name,
+        threshold: srmThreshold,
+      });
+    },
+  });
+
+export const notifyExperimentChange = async ({
+  context,
+  snapshot,
+}: {
+  context: Context;
+  snapshot: ExperimentSnapshotDocument;
+}) => {
+  const experiment = await getExperimentById(context, snapshot.experiment);
+  if (!experiment) throw new Error("Error while fetching experiment!");
+
+  if (
+    !context.org.settings?.experimentNotificationsEnabled &&
+    !experiment.metricsNotificationsEnabled
+  )
+    return;
+
+  const [
+    {
+      results: [lastResult],
```

Comment:
Get first result

Reply:
from first analysis, and we also want to only fire these notifications for a certain set of snapshots (has no dimension might be the only check we do for now).